### ISSUE TITLE
IDISP013 does not trigger for using declaration

### ISSUE THREAD

**Issue by Bouke** [DotNetAnalyzers/IDisposableAnalyzers#471]
A using statement does throw IDISP013 as expected:
```cs
using (var scope = CreateScope())
    return wrapped.HandleAsync(command, token);
```

However rewriting this to a using declaration does not trigger it, but should:
```cs
using var scope = CreateScope();
return wrapped.HandleAsync(command, token);
```

Failing test:
```cs

        [Test]
        public static void UsingDeclaration()
        {
            var code = @"
namespace N
{
    using System.Threading.Tasks;

    public static class C
    {
        public static ValueTask<int> MAsync()
        {
            using var file = System.IO.File.OpenRead(string.Empty);
            return ↓InnerAsync();
        }

        private static async ValueTask<int> InnerAsync()
        {
            await Task.Delay(10).ConfigureAwait(false);
            return 1;
        }
    }
}";
            RoslynAssert.Diagnostics(Analyzer, ExpectedDiagnostic, code);
        }
```

**Comment by Bouke** [comment]
Failing test:

```cs
        [Test]
        public static void UsingDeclaration()
        {
            var code = @"
namespace N
{
    using System.Threading.Tasks;

    public static class C
    {
        public static ValueTask<int> MAsync()
        {
            using var file = System.IO.File.OpenRead(string.Empty);
            return ↓InnerAsync();
        }

        private static async ValueTask<int> InnerAsync()
        {
            await Task.Delay(10).ConfigureAwait(false);
            return 1;
        }
    }
}";
            RoslynAssert.Diagnostics(Analyzer, ExpectedDiagnostic, code);
        }
```

**Comment by JohanLarsson** [comment]
Great issue, thanks!

**Comment by Bouke** [comment]
Thank you as well for your work on this analyzer. I've tried 4.0.4. on our codebase and it reported an issue, but (luckily?) it was a false positive: #473.
